R/utils.R

```diff
@@ -209,22 +181,30 @@ complete_cumulative_columns <- function(data) {
 #' then calculates the second from the first.
 #' @param data A data frame
 #' @return A data frame with extra columns if required
-#' @importFrom dplyr %>% mutate
+#' @importFrom dplyr %>% mutate group_by_at arrange vars starts_with lag
+#' @importFrom tidyr replace_na
 #' @importFrom tibble tibble
 #' 
 calculate_columns_from_existing_data <- function(data) {
   possible_counts <- c("cases", "deaths", "hosp", "recovered", "tested")
-
+  
   for (count in possible_counts) {
     count_today_name <- paste0(count, "_new")
     cumulative_count_name <- paste0(count, "_total")
 
     if (count_today_name %in% colnames(data) & !(cumulative_count_name %in% colnames(data))) {
       # in this case the daily count is there but there are no cumulative counts
-      data <- data %>% dplyr::mutate(!!cumulative_count_name := get_cumulative_from_daily(data[[count_today_name]]))
-    } else if (!(count_today_name %in% colnames(data)) & cumulative_count_name %in% colnames(data)) {
+      data <- data %>% 
+        dplyr::group_by_at(dplyr::vars(dplyr::starts_with("region_level"))) %>%
+        dplyr::arrange(date, .by_group = TRUE) %>%
+        dplyr::mutate(!!cumulative_count_name := cumsum(tidyr::replace_na(!!as.name(count_today_name), 0)))
+    
+      } else if (!(count_today_name %in% colnames(data)) & cumulative_count_name %in% colnames(data)) {
       # in this case the cumulative counts are there but no daily counts
-      data <- data %>% dplyr::mutate(!!count_today_name := get_daily_from_cumulative(data[[cumulative_count_name]]))
+      data <- data %>% 
+        dplyr::group_by_at(dplyr::vars(dplyr::starts_with("region_level"))) %>%
+        dplyr::arrange(date, .by_group = TRUE) %>%
+        dplyr::mutate(!!count_today_name := (!!as.name(cumulative_count_name)) - dplyr::lag(!!as.name(cumulative_count_name)))
```

Comment:
Might want fill = 0 to avoid NAs

Reply:
@kathsherratt what do you think?

Reply:
Thanks for spotting. Should be fixed in latest commit. 
Where we are calculating daily from cumulative data, I have replaced NAs with the last observation (since it is cumulative, to assume 0 change). 
(When calculating cumulative from daily, NAs are taken as 0 daily)